akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -135,6 +135,7 @@ abstract class SerializerWithStringManifest extends Serializer {
    * and message is dropped. Other exceptions will tear down the TCP connection
    * because it can be an indication of corrupt bytes from the underlying transport.
    */
+  @throws(classOf[NotSerializableException])
```

Comment:
@patriknw I still don't understand how this is bin/src-comp, doesn't it break all serialisers implemented in Java with the old api?

```
scala> class A { def m(): Unit = {} }
defined class A

scala> class B { @throws(classOf[Exception]) def m(): Unit = {} }
defined class B

scala> :javap A
...snip...
  public void m();
...snip...
scala> :javap B
...snip...
  public void m() throws java.lang.Exception;
...snip...
```